txnsync/service.go

```diff
@@ -51,6 +51,13 @@ func MakeTransactionSyncService(log logging.Logger, conn NodeConnector, isRelay
 	}
 	s.state.service = s
 	s.state.xorBuilder.MaxIterations = 10
+	if cfg.TxPoolSize > maxEncodedTransactionGroups {
```

Comment:
This looks correct, but think it would be cleaner if you'll move all that logic into a single method which would be called by both `init()` as well as from here.

Reply:
init methods are called when the package is loaded. In this case, instead of initializing the global variables where they are declared, call
```golang
func init() {
    setTransactionSyncVariables(config.GetDefaultLocal())
}
...
func setTransactionSyncVariables(cfg config.Local) {
        if cfg.TxPoolSize < maxEncodedTransactionGroups {
              return
        }
 	maxEncodedTransactionGroups = cfg.TxPoolSize
 	maxEncodedTransactionGroupEntries = cfg.TxPoolSize
 	maxBitmaskSize = (maxEncodedTransactionGroupEntries+7)/8 + 1
 	maxSignatureBytes = maxEncodedTransactionGroupEntries * len(crypto.Signature{})
 	maxAddressBytes = maxEncodedTransactionGroupEntries * crypto.DigestSize

 	maxBloomFilterSize = cfg.TxPoolSize * 5 // 32 bit xor uses slightly more than 4 bytes/element.
 	maxEncodedTransactionGroupBytes = cfg.TxPoolSize * 10000 // assume each transaction takes 10KB, as a worst-case-scenario for bounding purposes only.
 }
```